File: nutmilk-admin-dashboard/src/main/java/com/nutmilk/admin/service/CartService.java
package com.nutmilk.admin.service;

import com.nutmilk.admin.dto.cart.AddToCartRequest;
import com.nutmilk.admin.entity.Cart;

public interface CartService {
    Cart getCart(String username);
    Cart addToCart(String username, AddToCartRequest request);
    Cart removeFromCart(String username, Long cartItemId);
    void clearCart(String username);
}

File: nutmilk-frontend/src/services/cart.service.ts
import api from "./api";

const getCart = async () => {
    const response = await api.get("/cart");
    return response.data;
};

const addToCart = async (productId: number, quantity: number) => {
    const response = await api.post("/cart/add", { productId, quantity });
    return response.data;
};

const removeFromCart = async (cartItemId: number) => {
    const response = await api.delete(`/cart/remove/${cartItemId}`);
    return response.data;
};

const CartService = {
    getCart,
    addToCart,
    removeFromCart,
};

export default CartService;
